transformer_engine/pytorch/tensor/float8_blockwise_tensor.py

```diff
@@ -543,19 +543,79 @@ def forward(
         if shape is None:
             return tensor
 
-        if list(shape) != list(tensor.shape):
-            raise NotImplementedError("View not implemented.")
-        return tensor
+        # Canonicalize shape
+        if not isinstance(shape, Iterable):
+            shape = [shape]
+        elif len(shape) == 1 and isinstance(shape[0], Iterable):
+            shape = shape[0]
+        if -1 in shape:
+            shape = list(shape)
+            d_inferred = -math.prod(ctx.shape) // math.prod(shape)
+            for i, d in enumerate(shape):
+                if d == -1:
+                    shape[i] = d_inferred
+                    break
+        if shape[-1] != ctx.shape[-1]:
+            raise RuntimeError(
+                "Float8BlockwiseQTensor does not support reshaping inner dimension "
+                f"(attempted to reshape dims={tuple(tensor.shape)} to {tuple(shape)})"
+            )
+
+        if list(shape) == list(tensor.shape):
+            return tensor
+
+        if tensor._is_2D_scaled:
+            raise NotImplementedError("View of 2D scaled tensor is not implemented")
+
+        # Construct new tensor if shape is provided
+        new_rowwise_data = None
+        new_columnwise_data = None
+        if tensor._rowwise_data is not None:
+            new_rowwise_data = tensor._rowwise_data.view(*shape)
+        if tensor._columnwise_data is not None:
+            columnwise_shape = [shape[-1]] + list(shape[:-1])
+            new_columnwise_data = tensor._columnwise_data.view(columnwise_shape)
+
+        return Float8BlockwiseQTensor(
+            shape=shape,
+            dtype=tensor.dtype,
+            fp8_dtype=tensor._fp8_dtype,
+            rowwise_data=new_rowwise_data,
+            rowwise_scale_inv=tensor._rowwise_scale_inv,
+            columnwise_data=new_columnwise_data,
+            columnwise_scale_inv=tensor._columnwise_scale_inv,
+            quantizer=tensor._quantizer,
+            is_2D_scaled=tensor._is_2D_scaled,
+            requires_grad=tensor.requires_grad,
+        )
 
     @staticmethod
     def backward(
         ctx,
         grad: torch.Tensor,
     ) -> Tuple[Optional[torch.Tensor], ...]:
         # pylint: disable=missing-function-docstring
-
         if isinstance(grad, Float8BlockwiseQTensor):
-            raise NotImplementedError("View bwd not implemented")
+            new_data = (
+                grad._rowwise_data.view(*ctx.shape) if grad._rowwise_data is not None else None
+            )
+            if grad._columnwise_data is not None:
+                new_columnwise_data = grad._columnwise_data.view(ctx.shape[-1], -1)
```

Comment:
Why are we flattening the column-wise data to 2D?
```suggestion
                columnwise_shape = [ctx.shape[-1]] + list(ctx.shape[:-1])
                new_columnwise_data = grad._columnwise_data.view(columnwise_shape)
```

Reply:
I initially copied the code from the `MXFP8` tensor implementation. After reviewing the logic here, I believe the following approach is better. Could there be a bug in the `MXFP8` tensor?

```
if grad._is_2D_scaled:
    columnwise_shape = list(ctx.shape[-2:]) + list(ctx.shape[:-2])
else:
    columnwise_shape = [ctx.shape[-1]] + list(ctx.shape[:-1])
```

Additionally, I think the reshape/view operation for columnwise_data might be incorrect. Unlike the rowwise data, we cannot keep the inner dimension unchanged. We can only choose one  whose inner dim stay unchanged between rowwise data and columnwise tensor.  Perhaps it would be helpful to add an assertion or warning for tensors with columnwise tensor?